Update release process to use templating (#353)
kyma-project/community: guidelines/releases-guidelines/02-kyma-release-process.md

   > **NOTE:** If you don't create the Kyma release branch at this point and add a  `post-rel{release_version}-kyma-release-candidate` post-submit job to the `test-infra` master, then pushing anything to the Kyma release branch, creating or rebasing the branch, triggers a new GitHub release.

2. [Define new release jobs](#kyma-release-process-kyma-release-process-preparation-define-new-release-jobs) in the `test-infra` repository.

### Define new release jobs

> 1. Navigate to the `test-infra` repository.

Reviewer: ```suggestion
1. Go to the `test-infra` repository.
```

2. Open `templates/config.yaml`

> 3. Change `global.releases` to contain new release. Also remove oldest release on the list.

Reviewer: ```suggestion
3. Add the new release to `global.releases`. Remove the oldest release on the list.
```

> 4. Change `global.nextRelease` to contain version for future release. 

Reviewer: ```suggestion
4. Set `global.nextRelease` to the future release version. 
```

5. Run `dep ensure -v --vendor-only` in the `development/tools` directory to install dependencies.
5. Run `go run development/tools/cmd/rendertemplates/main.go --config templates/config.yaml` in the root of the repository to generate jobs.

> 6. Run `go test ./development/tools/jobs/...` in the root of the repository. If anything is red fix it using these guidelines: 

Reviewer: ```suggestion
6. Run `go test ./development/tools/jobs/...` in the root of the repository. If any of the tests is marked red, fix it using these guidelines: 
```

>   * For release tests using `GetKymaReleasesSince` with a release that is no longer supported change it to use `GetAllKymaReleases`.

Reviewer: ```suggestion
  * For release tests using `GetKymaReleasesSince` with a release that is no longer supported, change the method to `GetAllKymaReleases`.
```

>   * For release tests using `GetKymaReleasesUntil` with a release that is no longer supported simply remove that part of the test.

Reviewer: ```suggestion
  * For release tests using `GetKymaReleasesUntil` with a release that is no longer supported, remove the part of the test which includes the method.
```

> 7. When tests are green commit all jobs and you just got yourself new release jobs.

Reviewer: ```suggestion
7. If tests are green, commit all jobs. The new release jobs are ready.
```

### Define new release jobs

> 

Reviewer 2: You can remove this step from the preparation list above.

Author: Done
